imodqgis/cross_section/cross_section_data.py

```diff
@@ -134,6 +132,43 @@ def edit_colors(self):
             self.colors_changed.emit()
 
 
+class StaticOnlyMixin():
+    def requires_loading(self, **kwargs) -> bool:
+        return self.x is None
+
+class SupportsTemporalMixin():
+    def requires_loading(self, datetime_range: QgsDateTimeRange) -> bool:
+        group_index = self.variables_indexes[self.variable][self.layer_numbers[0]]
+        if self.requires_static_index(
+            datetime_range
+        ):  # Just take the first one in such a case
+            sample_index = (0, group_index)
+        else:
+            index = self.layer.datasetIndexAtTime(datetime_range, group_index)
+            sample_index = (index.dataset(), index.group())
+
+        if sample_index == self.sample_index:
```

Comment:
I probably choose `sample_index` myself as a name, but can we think of a better name? ...

Maybe `time_and_group_index`?